pkg/stanza/docs/operators/recombine.md

```diff
@@ -17,7 +17,7 @@ The `recombine` operator combines consecutive logs into single logs based on sim
 | `max_unmatched_batch_size`     | 100                        | The maximum number of consecutive entries that will be combined into a single entry before the match occurs (with `is_first_entry` or `is_last_entry`), e.g. `max_unmatched_batch_size=0` - all entries combined, `max_unmatched_batch_size=1` - all entries uncombined until the match occurs, `max_unmatched_batch_size=100` - entries combined into 100-entry-packages until the match occurs  |
 | `overwrite_with`               | `newest`                   | Whether to use the fields from the `oldest` or the `newest` entry for all the fields that are not combined. |
 | `force_flush_period`           | `5s`                       | Flush timeout after which entries will be flushed aborting the wait for their sub parts to be merged with. |
-| `source_identifier`            | `$attributes["file.path"]` | The [field](../types/field.md) to separate one source of logs from others when combining them. |
+| `source_identifier`            |                            | The [field](../types/field.md) to separate one source of logs from others when combining them. |
```

Comment:
The default is still `attributes["file.path"]` (though the `$` should be removed).

```suggestion
| `source_identifier`            | `attributes["file.path"]`. | The [field](../types/field.md) to separate one source of logs from others when combining them. |
```

However, the default appears to be out of date with the semantic convention, which is [`log.file.path`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/57a5d0d350f1b6cb08cd0efc5952425d0353d8d6/pkg/stanza/fileconsumer/attrs/attrs.go#L15), so we should change the default in another PR.

The `"DefaultSourceIdentifier"` is the default _value_ used when the field is not present on a log entry.

Reply:
Hummm, I see, @djaglowski. I think I got confused because of the warning in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/operator/transformer/recombine/transformer.go#L119.

So the "entry" in this case means the log entry being processed, not the configuration of that recombine, right? 

Isn't it weird that a log entry didn't have a log file path? 🤔

Reply:
Oh oh, fixing this would be a breaking change. Is this change desired right now? I'm open to do it.

Reply:
> So the "entry" in this case means the log entry being processed, not the configuration of that recombine, right?

Exactly.

> Oh oh, fixing this would be a breaking change, right? Is this change desired right now? I'm open to do it.

Yes, unfortunately, but I think it's one we ought to do. The intention was always that logs would be recombined by default according to the files they were read from. Unfortunately it seems we updated the attribute name in one place and not the other so it's likely not working as intended for many users. Arguably this is just a bug fix, but since it's been in place for so long I think we should mark it as breaking.

Reply:
@djaglowski I updated the PR given the outcome of our conversation, including its title. Please have a look whenever you have some time, thank you!